### PR TITLE
CC firefly product page UI, search integration with firefly web tests

### PR DESCRIPTION
**Pull request by sigadamvenkata**
CC firefly product page UI, search integration with firefly web tests

### CODE REVIEW

**Review by JackySun9 — APPROVED** [review]

File: selectors/cc/firefly.page.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   */
+
+  async stage_login(email, password) {
+    expect(process.env.IMS_EMAIL, 'ERROR: No environment variable found for IMS_EMAIL').toBeTruthy();
+    expect(process.env.IMS_PASS, 'ERROR: No environment variable found for IMS_PASS.').toBeTruthy();
+    console.log('using IMS stage credentials to login to Firefly custom IMS page');
+    // Wait for page to load & stabilize:
+    await this.page.waitForLoadState('networkidle');
+    // Wait for Firefly IMS page
+    await this.IMSPageLoad.waitFor({ state: 'visible', timeout: 5000 });
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:37]: You don't need this wait. Playwright's fill function does auto-waiting which will check that the email field is visible, enabled, and editable. https://playwright.dev/docs/actionability

File: selectors/cc/firefly.page.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    await this.signInEmailAddress.fill(email);
+    await this.page.keyboard.press('Tab');
+    await this.emailIdContinueBtn.click();
+    // Wait for next IMS page to load
+    await this.page.waitForLoadState('domcontentloaded');
+    // Enter password and proceed
+    await this.IMSPasswordPage.waitFor({ state: 'visible', timeout: 10000 });
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:44]: You don't need this wait. Playwright's fill function does auto-waiting which will check that the email field is visible, enabled, and editable. https://playwright.dev/docs/actionability

File: selectors/cc/firefly.page.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    // await this.signInPassword.waitFor({ state: 'visible', timeout: 5000 });
+    await this.signInPassword.fill(password);
+    // Complete the login flow:
+    await this.passwordContinueBtn.waitFor({ state: 'visible', timeout: 10000 });
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:48]: You don't need this wait. Playwright's click function does auto-waiting which will check that the email field is visible, enabled, and editable. https://playwright.dev/docs/actionability

File: selectors/cc/firefly.page.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    await this.passwordContinueBtn.click();
+    // Check if login process was successful:
+    await this.FFpageLoad.waitFor({ state: 'visible', timeout: 20000 });
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:51]: You could use 
`await expect(this.FFpageLoad).toBeVisible();
`
Which has auto-retrying assertions. https://playwright.dev/docs/test-assertions#expectlocatortobevisibleoptions

File: selectors/cc/firefly.page.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    console.info('Login success');
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  async promtbar_input(searchtext) {
+    console.log('this function use check give prompt text reside in product page');
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:57]: Sounds more like a comment?

File: selectors/cc/firefly.page.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    // eslint-disable-next-line no-template-curly-in-string
+    this.promptsearchtext = this.page.locator('//textarea[contains (text(),"#${searchtext}")]');
```
Comment (Dli3) [inline selectors/cc/firefly.page.js:59]: `    const promptsearchtext = this.page.locator('//textarea[contains (text(),"#${searchtext}")]');
`

File: tests/cc/firefly.test.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  });
+  test(`${features[0].name},${features[0].tags}`, async ({ page, baseURL }) => {
+    console.info(`[Test Page]: ${baseURL}${features[0].path}`);
+    await test.step('Firefly interactive marquee search via IMS login', async () => {
+      await page.goto(`${baseURL}${features[0].path}`);
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[0].path}`);
+    });
+    await test.step('search term take use to IMS and post login to FF product page', async () => {
+      await page.waitForLoadState();
```
Comment (Dli3) [inline tests/cc/firefly.test.js:18]: You're waiting for loadstate on line 14, you probably don't need another one.

File: tests/cc/firefly.test.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      const searchtext = features[0].term;
+      expect(await firefly.searchPromptBox).toBeTruthy();
+      expect(await firefly.generateCTA).toBeTruthy();
+      await firefly.searchPromptBox.fill(features[0].term);
+      // await page.waitForTimeout(2000);
```
Comment (Dli3) [inline tests/cc/firefly.test.js:23]: You can probably remove this comment if not needed.

File: tests/cc/firefly.test.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+  test(`${features[1].name},${features[1].tags}`, async ({ page, baseURL }) => {
+    console.info(`[Test Page]: ${baseURL}${features[1].path}`);
+    await test.step('click generate without search prompt', async () => {
+      await page.goto(`${baseURL}${features[1].path}`);
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[1].path}`);
+    });
+    await test.step('user hits serach CTA with out search prompt', async () => {
+      await page.waitForLoadState();
```
Comment (Dli3) [inline tests/cc/firefly.test.js:39]: You're waiting for loadstate on line 35, you probably don't need another one.

File: tests/cc/firefly.test.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  });
+
+  test(`${features[2].name},${features[2].tags}`, async ({ page, baseURL }) => {
+    console.info(`[Test Page]: ${baseURL}${features[2].path}`);
+    await test.step('click generate without search prompt', async () => {
+      await page.goto(`${baseURL}${features[2].path}`);
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[2].path}`);
+    });
+    await test.step('user hits serach CTA with out search prompt', async () => {
```
Comment (Dli3) [inline tests/cc/firefly.test.js:54]: `    await test.step('user hits search CTA with out search prompt', async () => {
`
Typo fix.

File: tests/cc/firefly.test.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+      await page.waitForLoadState();
```
Comment (Dli3) [inline tests/cc/firefly.test.js:55]: You're waiting for loadstate on line 51, you probably don't need another one.

**Pull request merged**